### ISSUE TITLE
Rule proposal: enforce shape of dangerouslySetInnerHTML

### ISSUE THREAD

**Issue by lencioni** [yannickcr/eslint-plugin-react#711]
React will issue a warning if you try to use dangerouslySetInnerHTML with a value of the wrong shape.

https://github.com/facebook/react/blob/85dcbf83/src/renderers/dom/shared/ReactDOMComponent.js#L173-L177

This should be easy enough to enforce for most cases with a linter rule.

Bad:

```jsx
<div dangerouslySetInnerHTML={{ }} />
```

```jsx
<div dangerouslySetInnerHTML={{ html: 'HTML' }} />
```

```jsx
<div dangerouslySetInnerHTML={true} />
```

Good:

```jsx
<div dangerouslySetInnerHTML={{ __html: 'HTML' }} />
```

**Comment by ljharb** [comment]
This seems like it should perhaps be an addition to `no-danger`?

**Comment by lencioni** [comment]
Perhaps, but I think that would require renaming no-danger to something more general, and giving it options.

I like keeping them separate, because someone might want to generally enable no-danger, but disable it in a specific situation and still have this protection.
